src/request.js

```diff
@@ -37,6 +37,7 @@ module.exports = class Request extends EventEmitter {
   handle: ?any; // TODO: Figure out the type here.
   error: ?Error;
   connection: ?any; // TODO: This should be `Connection`, not `any`.
+  timeout: ?number;
```

Comment:
I think the correct type to use here would be:

```js
timeout: number | typeof undefined;
```

`?number` is `number | null | typeof undefined;`, but we shouldn't allow `null` here, but let's not add another incorrect type here.